### PR TITLE
Code Quality: Introduced view models to widgets

### LINKED ISSUE #13210: Code Quality: Refactor Widget controls
### Description

As #4180 lists this as an issue, widget controls need to be refactored.

Detailed issues:
- Home page is just wrapping WidgetsListControl.
- Each widget control is handling its all functions
- Drive widget won't be updated when drive info is updated 

Keep in mind:
- #5135
- #5136
- #4262
- #6182

### Concerned code

- UserControls.Widgets
- ViewModels.Widgets
- HomeViewModel
- HomePage

### Gains

- A better readability & maintainability.

### Requirements

- Refactor all concened code

### Comments

_No response_

### PR DESCRIPTION
**Pull request by 0x5bfa**
## Summary

- [x] Introduced view models to widgets
- [x] Simplify the widget codebase
- [x] Replaced ICommands with RichCommands
  - [x] EjectDrive
  - [x] FormatDrive
  - [x] Pin
  - [x] Unpin
  - [x] OpenInNewPane
  - [x] OpenInNewTab
  - [x] OpenInNewWindow
  - [ ] OpenProperties
  - [x] OpenFileLocation
  - [x] RemoveRecentItem
  - [x] ClearRecentItem
- [x] Make Refresh button at work

## Details

C#:
- Moved code behind codes to newly created view models
- Removed a bunch of event handlers in Home page and let handle events in each widget
- Removed WidgetsHelper because it's now useless
- Removed duplicated codes and unified
- Replace ICommands with IRichCommands

XAML:
- No UI modifications
- Moved resources xaml out of HomePage and placed to newly created ResourceDictionary

## PR Checklist

- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #13210
- [x] Did you build the app and test your changes?
- [x] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Lanuch the app
   2. See the widgets shown correctly
   3. Right click an item
   4. See the context flyout shown correctly
   5. Add a new tab
   6. See the Home page loaded correctly

Tested locally ✅

### CODE REVIEW

File: src/Files.Core/Data/Enums/WidgetSectionType.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) 2023 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+namespace Files.Core.Data.Enums
+{
+	public enum WidgetSectionType
+	{
+		QuickAccess,
+		Drive,
```
Comment (yaira2) [inline src/Files.Core/Data/Enums/WidgetSectionType.cs:9]:
```suggestion
		Drives,
```

**Comment by yaira2** [comment]
- Are the ViewModels shared or does each tab have its own instance?
- Does this fix the issue where refreshing the home page doesn't update the drives?

**Comment by 0x5bfa** [comment]
I can fix those in this PR. but this PR's purpose is to just introduce view models and to introduce RichCommands instead of ICommands.

> Are the ViewModels shared or does each tab have its own instance?

To achieve this goal we have to use single instance of HomeViewModel and should not use a list(ItemsRepeater) for widget UserControls but expand instead.

> Does this fix the issue where refreshing the home page doesn't update the drives?

I think yes, yet I'm gonna test again. When I simplify methods, I made the refresh method to be reused on loaded and reloaded.

File: src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml [hunk 2]
```diff
@@ -57,11 +56,8 @@
 			<ListView.ItemTemplate>
 				<DataTemplate x:DataType="recent:RecentItem">
 					<Grid
-						Padding="2.5"
-						HorizontalAlignment="Stretch"
-						VerticalAlignment="Stretch"
 						AutomationProperties.Name="{x:Bind Name}"
-						ColumnSpacing="14"
+						ColumnSpacing="16"
```
Comment (yaira2) [inline src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml:60]: > No UI modifications

File: src/Files.Core/Data/Enums/WidgetSectionType.cs (hunk 1 above)
Comment (0x5bfa) [reply src/Files.Core/Data/Enums/WidgetSectionType.cs:9]: Thanks!

File: src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml (hunk 2 above)
Comment (0x5bfa) [reply src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml:60]: Oh, missed. Thanks!

Comment (0x5bfa) [reply src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml]: Done

File: src/Files.Core/Data/Enums/WidgetSectionType.cs (hunk 1 above)
Comment (0x5bfa) [reply src/Files.Core/Data/Enums/WidgetSectionType.cs]: Done

**Comment by 0x5bfa** [comment]
> Are the ViewModels shared or does each tab have its own instance?

When I tried last time, memory leak would occur in HomePage. Perhaps, HomePage should use each UserControl instead of ItemsRepeater. I'll find a way into.

> Does this fix the issue where refreshing the home page doesn't update the drives?

I found that the event HomePage hooked has never been invoked by SidebarViewModel. This means clicking Refresh on Home page doesn't make any action. I will re-hook.

**Pull request closed without merging**